### ISSUE TITLE
eslint compile time error - Environment key "vue/setup-compiler-macros" is unknown

### ISSUE THREAD

**Issue by maxweld** [vuejs/eslint-plugin-vue#1695]
**Checklist**

- [x ] I have tried restarting my IDE and the issue persists.
- [x ] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:  6.7.2** 
- **eslint-plugin-vue version: 7.0.0** 
- **Node version: 14.17.6** 
- **Operating System: Windows 10 pro** 

**Please show your full configuration:**

```json5
module.exports = {
  root: true,

  env: {
    'vue/setup-compiler-macros': true,
    node: true,
  },

  extends: [
    'plugin:vue/vue3-essential',
    'eslint:recommended',
    '@vue/typescript/recommended',
  ],

  parserOptions: {
    ecmaVersion: 2020,
  },

  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
  },

  extends: [
    'plugin:vue/vue3-recommended',    // also tried -essential
    'eslint:recommended',
    '@vue/typescript/recommended',
  ],
}
```

**What did you do?**	

I am building a modified sample app using Vue 3.2, TypeScript and using the new <script setup lang=ts> capability.
It has been working to date, but I have just introduce components passing Props and thus introduced the defineProps compiler function. Reading the FAQ, I introduced the 
`'vue/setup-compiler-macros': true,`
to the env section of the eslintrc.js file.
This seemed to work at first, but then started giving an error at compile time. This happens both in VS Studio on save, or if I run `npm run serve`
in a command window outside VS Code.

**Note 1:** While investigating the error, I identified that there are two 'extends' sections in the eslint config file. This was build by the Vue CLI.  I removed the second one to see if this made any difference, but it did not. I also tried using the -essential in both places

**Note 2:** I am relatively inexperienced using TypeScript in Vue, and am trying to pass props etc. So I may be doing something wrong in this respect.

***Actual Source Code - TodoFilters.vue***
```vue
<script setup lang="ts">
const props = defineProps({
  type: String,
  remainingTodos: Number,
  filterResults: Function,
  filterType: String,
  clearCompleted: Function,
})
</script>

<template>
  <div class="footerList">
    <span>{{ remainingTodos }} items </span>
    <ul>
      <li v-on:click="filterResults!('all')">
        <a v-if="filterType === 'all'" class="selected">All</a>
        <a v-else>All</a>
      </li>
      <li v-on:click="filterResults!('active')">
        <a v-if="filterType === 'active'" class="removePaddLeft selected"
          >Active</a
        >
        <a v-else class="removePaddLeft">Active</a>
      </li>
      <li v-on:click="filterResults!('completed')">
        <a v-if="filterType === 'completed'" class="removePaddLeft selected"
          >Completed</a
        >
        <a v-else class="removePaddLeft">Completed</a>
      </li>
    </ul>
    <button
      v-if="type === 'private'"
      class="clearComp"
      v-on:click="clearCompleted!"
    >
      Clear Completed
    </button>
  </div>
</template>

```

**What did you expect to happen?**
I expected it to compile cleanly or present some error that I could understand.

**What actually happened?**

The error is
`Syntax Error: Error: .eslintrc.js:
        Environment key "vue/setup-compiler-macros" is unknown
    at Array.forEach (<anonymous>)`

I also get an error trace in the browser ...
```
Failed to compile.

./src/components/TodoFilters.vue
Module build failed (from ./node_modules/eslint-loader/index.js):
Error: .eslintrc.js:
	Environment key "vue/setup-compiler-macros" is unknown`
```

    at C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint\lib\shared\config-validator.js:169:19
    at Array.forEach (<anonymous>)
    at validateEnvironment (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint\lib\shared\config-validator.js:163:30)
    at validateConfigArray (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint\lib\shared\config-validator.js:334:9)
    at CascadingConfigArrayFactory._finalizeConfigArray (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js:417:13)
    at CascadingConfigArrayFactory.getConfigArrayForFile (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js:271:21)
    at CLIEngine.isPathIgnored (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint\lib\cli-engine\cli-engine.js:951:18)
    at CLIEngine.executeOnText (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint\lib\cli-engine\cli-engine.js:868:38)
    at lint (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint-loader\index.js:278:17)
    at transform (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\eslint-loader\index.js:252:18)
    at C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\loader-fs-cache\index.js:127:18
    at ReadFileContext.callback (C:\Users\david\Documents\My-Computing-Projects\Hasura\vue3-ts-selfbuild\node_modules\loader-fs-cache\index.js:31:14)
    at FSReqCallback.readFileAfterOpen [as oncomplete] (fs.js:282:13)`

**Repository to reproduce this issue**

https://github.com/maxweld/vue3-ts-selfbuild

**Comment by maxweld** [comment]
current workaround is to use 
```
  globals: {
    defineProps: 'readonly',
    defineEmits: 'readonly',
  },
```
in eslintrc.js

**Comment by ota-meshi** [comment]
It feature were added in v8. Please use the latest version.

**Comment by DrakoPOD** [comment]
@ota-meshi I have the same issue, but I don't know what I need to update
